header_timing_test.go

```diff
@@ -26,6 +26,72 @@ func (r *benchReadBuf) Read(p []byte) (int, error) {
 	return n, nil
 }
 
+type oldHeaderScanner struct {
+	headerScanner
+}
+
+func (m *oldHeaderScanner) next() bool {
+	bLen := len(m.b)
+	if bLen >= 2 && m.b[0] == '\r' && m.b[1] == '\n' {
+		m.b = m.b[2:]
+		m.hLen += 2
+		return false
+	}
+	if bLen >= 1 && m.b[0] == '\n' {
+		m.b = m.b[1:]
+		m.hLen++
+		return false
+	}
+	n := bytes.IndexByte(m.b, ':')
+	if n < 0 {
+		m.err = errNeedMore
+		return false
+	}
+	m.key = m.b[:n]
+	normalizeHeaderKey(m.key, m.disableNormalizing)
+	n++
+	for len(m.b) > n && m.b[n] == ' ' {
+		n++
+	}
+	m.hLen += n
+	m.b = m.b[n:]
+	n = bytes.IndexByte(m.b, '\n')
+	if n < 0 {
+		m.err = errNeedMore
+		return false
+	}
+	m.value = m.b[:n]
+	m.hLen += n + 1
+	m.b = m.b[n+1:]
+
+	if n > 0 && m.value[n-1] == '\r' {
+		n--
+	}
+	for n > 0 && m.value[n-1] == ' ' {
+		n--
+	}
+	m.value = m.value[:n]
+	return true
+}
+
+func BenchmarkHeaderMultiLineValueOld(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		hs := new(oldHeaderScanner)
+		hs.b = webLogicResponse[17:]
+		for hs.next() {
+		}
+	}
+}
```

Comment:
I don't think we should keep this old code here just for a benchmark comparison. If you can run the benchmark locally and add the result as a comment to this pull request it's good enough. Then you can remove this benchmark.